controllers/controllers.go

```diff
@@ -155,6 +155,31 @@ func (r *instanceReconciler) deconfigureInstance(node *core.Node) error {
 	return nil
 }
 
+// windowsNodeUpgradePredicate returns a predicate whose filter catches only Windows node objects that signal upgrade
+// to the latest WMCO version through a change in the node version annotation
+func windowsNodeUpgradePredicate() predicate.Funcs {
```

Comment:
This isnt actually catching upgrades
this is just filtering out outdated nodes
any time a fully upgraded node changes state this predicate will allow the event

please rename this function